lib/rubygems/safe_marshal/reader.rb

```diff
@@ -88,31 +88,51 @@ def read_element
         when 85 then read_user_marshal # ?U
         when 91 then read_array # ?[
         when 102 then read_float # ?f
-        when 105 then Elements::Integer.new int: read_integer # ?i
+        when 105 then Elements::Integer.new(read_integer) # ?i
         when 108 then read_bignum # ?l
         when 111 then read_object # ?o
         when 117 then read_user_defined # ?u
         when 123 then read_hash # ?{
         when 125 then read_hash_with_default_value # ?}
-        when "e".ord then read_extended_object
-        when "c".ord then read_class
-        when "m".ord then read_module
-        when "M".ord then read_class_or_module
-        when "d".ord then read_data
-        when "/".ord then read_regexp
-        when "S".ord then read_struct
-        when "C".ord then read_user_class
+        when 101 then read_extended_object # ?e
+        when 99 then read_class # ?c
+        when 109 then read_module # ?m
+        when 77 then read_class_or_module # ?M
+        when 100 then read_data # ?d
+        when 47 then read_regexp # ?/
+        when 83 then read_struct # ?S
+        when 67 then read_user_class # ?C
         else
           raise Error, "Unknown marshal type discriminator #{type.chr.inspect} (#{type})"
         end
       end
 
+      STRING_E_SYMBOL = Elements::Symbol.new("E").freeze
+      private_constant :STRING_E_SYMBOL
+
       def read_symbol
-        Elements::Symbol.new name: @io.read(read_integer)
+        len = read_integer
+        if len == 1
+          byte = read_byte
+          if byte == 69 # ?E
+            STRING_E_SYMBOL
```

Comment:
I saw A and at least one other mentioned in tests. Are the others just test cases and E is the only really common one?

Reply:
`E` is the only one that is allowlisted, but I wanted to be comprehensive in my tests.

`E` is the symbol used for string encoding, so it shows up very often